website/docs: updated security release procedure (#15288)
goauthentik/authentik: website/docs/developer-docs/releases/index.md

> 
>     Include the new file in the `/website/sidebars.js`
> 
>     Push the branch to https://github.com/goauthentik/authentik-internal for CI to run and for reviews
> 
>     An image with the fix is built under `ghcr.io/goauthentik/internal-server` which can be made accessible to the reporter for testing
> 
> - Check with the original reporter that the fix works as intended
> - Wait for GitHub to assign a CVE
> - Announce the release of the vulnerability via Mailing list and discord

Reviewer: GitHub won't allow me to make the same suggestion for the above bullet points.

```suggestion

    - Include the new file in the `/website/sidebars.js`.

    - Push the branch to https://github.com/goauthentik/authentik-internal for CI to run and for reviews.

    - An image with the fix is built under `ghcr.io/goauthentik/internal-server` which can be made accessible to the reporter for testing.

- Check with the original reporter that the fix works as intended.
- Wait for GitHub to assign a CVE.
- Announce the release of the vulnerability via Mailing list and discord.
```

### Creating a security release

- On the date specified in the announcement, retag the image from `authentik-internal` to the main image:

    ```
    docker buildx imagetools create -t ghcr.io/goauthentik/server:xxxx.x ghcr.io/goauthentik/internal-server:gh-cve-2022-xxx
    docker buildx imagetools create -t ghcr.io/goauthentik/server:xxxx.x.x ghcr.io/goauthentik/internal-server:gh-cve-2022-xxx
    ```

>     Where xxxx.x is the version family and xxxx.x.x is the full version.

Reviewer: ```suggestion
    Where `xxxx.x` is the version family and `xxxx.x.x` is the full version.
```

    This will make the fixed container image available instantly, while the full release is running on the main repository.

> - Push the local `security/CVE-2022-xxxxx` branch into a PR, and squash merge it if the pipeline passes
> - If the fix made any changes to the API schema, merge the PR to update the web API client
> - Cherry-pick the merge commit onto the version branch
> - If the fix made any changes to the API schema, manually install the latest version of the API client in `/web`

Reviewer: ```suggestion
- Push the local `security/CVE-2022-xxxxx` branch into a PR, and squash merge it if the pipeline passes.
- If the fix included changes to the API schema, merge the PR to update the web API client.
- Cherry-pick the merge commit onto the version branch.
- If the fix included changes to the API schema, manually install the latest version of the API client in `/web`.
```
